addons/mrp/models/mrp_workorder.py

```diff
@@ -159,6 +160,19 @@ def _onchange_finished_lot_id(self):
             if line:
                 self.qty_producing = line.qty_done
 
+    @api.onchange('date_planned_start', 'date_planned_finished')
+    def _onchange_planned_dates(self):
+        if self.date_planned_start and self.date_planned_finished:
+            diff = self.date_planned_finished - self.date_planned_start
+            self.duration_expected = diff.total_seconds() / 60
+
+    @api.onchange('duration_expected')
+    def _onchange_duration_expected(self):
+        if self.date_planned_finished:
+            minutes = floor(self.duration_expected)
+            seconds = round((self.duration_expected % 1) * 60)
+            self.date_planned_finished = self.date_planned_start + relativedelta(minutes=minutes, seconds=seconds)
```

Comment:
you need to rebase because we introduce an onchange on date_planned_start here https://github.com/odoo/odoo/commit/2a2c27fa45d2a9859cf3714dc5c7d66cf0022e33#diff-f3cb8065b281bb0008cfd0f8ab0cdac5

maybe you can re-use this one?

Reply:
Both onchange will sync perfectly and no need of newly introduced [onchange](https://github.com/odoo/odoo/commit/2a2c27fa45d2a9859cf3714dc5c7d66cf0022e33#diff-6a737809e76574ef2f3d62898335762bR149). So we have to remove it.

Reply:
@jad-odoo when you change the `date_planned_start`, we must update the `date_planned_finished` accordling the `duration_expected`, not update the `duration_expected`.

You must update the `duration_expected` when you modify the `date_planned_finished`, not the `date_planned_start`.

Reply:
@jat-odoo Please see this comment that was sent to me because of a typo

Reply:
@jat-odoo What I said isn't in contradiction with your point 5.
* If you change `duration_expected`, you update the `date_planned_finished` (point 5 of your spec).
* If you change `date_planned_start`, you update the `date_planned_finished` ([OMA's review 7 august](https://www.odoo.com/web#id=1935229&action=333&active_id=966&model=project.task&view_type=form&menu_id=4720), "*Add an onchange on date_planned_start on mrp.workorder --> When I update date start, date end should be automatically updated too.*"), so you don't update `duration_expected` in this case otherwise you can't adapt the `date_planned_finished`.

So:
- Keep the old onchange to update the `date_planned_finished` and edit it to include the onchange on `duration_expected`.
- Add a new onchange to update the `duration_expected` **only** when you modify the `date_planned_finished`.

cc @oma-odoo

Reply:
@svs-odoo I got your points.
Here is conflict you need to know https://github.com/odoo/odoo/commit/16e7a4b4c74f2a391c033d3c732f542cb56e768d (same as your older onchange)

Reply:
Oh ,ok! I didn't know we had two onchange for the date, I understand better the situation now.
So, I'm OK with your changes here :+1: